forms/Form.php

```diff
@@ -1070,11 +1070,24 @@ public function getStrictFormMethodCheck() {
 	public function FormAction() {
 		if ($this->formActionPath) {
 			return $this->formActionPath;
-		} elseif($this->controller->hasMethod("FormObjectLink")) {
-			return $this->controller->FormObjectLink($this->name);
+		}
+
+		// Forms without parent controller have no link;
+        // E.g. Submission handled via graphql
+		if (!$this->controller) {
+			return null;
+		}
```

Comment:
Unlike 4.0, `$controller` is a required constructor arg in 3.x. Is this only reachable with:

```php
$form = new Form($this ...);
$form->setController(null);
```
??

Also might want to remove the reference to graphql, as it has no compatibility with 3.0 and likely never will.

Reply:
Ah right, that was a blind copy paste sorry. You're right.